crates/sui-types/src/move_package.rs

```diff
@@ -154,9 +242,82 @@ impl MovePackage {
                 })
                 .collect(),
             max_move_package_size,
+            type_origin_table,
+            linkage_table,
         )
     }
 
+    fn build_linkage_table(
+        dependencies: &[MovePackage],
+    ) -> Result<BTreeMap<ObjectID, UpgradeInfo>, ExecutionError> {
+        // all transitive dependencies are included so simply put them into the table
+        let mut linkage_table = BTreeMap::new();
+        for p in dependencies {
+            let bytes: &Vec<u8> =
+                p.module_map.values().next().expect(
+                    "Tried to build a Move package from an empty iterator of Compiled modules",
+                );
+            let module = CompiledModule::deserialize(bytes)
```

Comment:
Note, the max supported bytecode format is part of the protocol config, so we should be heeding that here.  Technically not going to be an issue/blocker for the cut on Monday because we don't have plans to update the bytecode format straight away, but we should fast follow with a fix for that.